java/daikon/dcomp/DCInstrument24.java

```diff
@@ -0,0 +1,4916 @@
+      MethodGen24 mgen,
+      MethodGen24.MInfo24 minfo,
+      List<CodeElement> instructions) {
+
+    List<CodeElement> newCode = createTagFrame(mgen);
+
+    // The start of the list of CodeElements looks as follows:
+    //   LocalVariable declarations (if any)
+    //   Label for start of code (if present)
+    //   LineNumber for start of code (if present)
+    //   <the actual code for the method>
+    //
+    // We want to insert the tag_frame setup code after the LocalVariables (if any) and after the
+    // inital label (if present), but before any LineNumber or Instruction.
+    ListIterator<CodeElement> li = instructions.listIterator();
+    CodeElement inst;
+    try {
+      while (li.hasNext()) {
+        inst = li.next();
+        if ((inst instanceof LineNumber) || (inst instanceof Instruction)) {
+          break;
+        }
+      }
+
+      // Insert the TagFrame code before the LineNumber or Instruction we just located.
+      // Back up the iterator to point to just before 'inst', then copy the newCode.
+      li.previous();
+      for (CodeElement ce : newCode) {
+        li.add(ce);
+      }
+
+      // We want to remember the current location in the instruction list. It is where a call to the
+      // DynComp runtime enter routine should be inserted, if we are tracking this method.
+      // This will also be used to indicate where the newStartLabel should be defined.
+      // Finally, we will also return to this location for code instrumentation after calculating
+      // the operand stack values.
+      newStartIndex = li.nextIndex();
+
+      // The next section of code calculates the operand stack value(s) for the current method.
+      if (debugOperandStack) {
+        System.out.printf("%nStarting operand stack calculation%n");
+      }
+
+      // Build a map of labels to instruction list offsets
+      labelIndexMap = new HashMap<>();
+      li = instructions.listIterator();
+      while (li.hasNext()) {
+        inst = li.next();
+        if (inst instanceof LabelTarget lt) {
+          if (debugOperandStack) {
+            System.out.println("label target: " + lt.label() + ", index: " + li.previousIndex());
+          }
+          // remember where this label is located within the instruction list
+          labelIndexMap.put(lt.label(), li.previousIndex());
+        }
+      }
+      if (oldStartLabel != null) {
+        // change offset of oldStartLabel to where we will define newStartLabel
+        labelIndexMap.put(oldStartLabel, newStartIndex);
+      }
+      labelIndexMap.put(newStartLabel, newStartIndex);
+
+      // Create an array to hold the calculated operand stack
+      // prior to each byte code instruction.
+      stacks = new OperandStack24[instructions.size()];
+
+      // Create an array containing the type of each local variable.
+      // This will be indexed by the local variable's slot number.
+      // There may be a gap for the second slot of a variable of type long or double.
```

Comment:
Does this mean that slot numbers are not sequential?  If so, they sound more like slot offsets than slot numbers or slot indexes.